core/commands/generate_product_repositories_command.rb

```diff
@@ -222,6 +223,10 @@ def write_repository(product_dir, releases)
     end
   end
 
+  def setup_release_architecture(release)
```

Comment:
Please inline this method inside the write_repository method. This method is bad:
* it has bad naming
* it modifies the argument!